dbt/adapters/redshift/impl.py

```diff
@@ -17,6 +19,76 @@ def type(cls):
     def date_function(cls):
         return 'getdate()'
 
+    @classmethod
+    def get_redshift_credentials(cls, config):
+        result = config.copy()
+
+        method = result.get('method')
+
+        if method == 'database':
+            return (result)
+
+        elif method == 'iam':
+            cluster_id = result.get('cluster_id')
```

Comment:
can you slide the logic in this `else` branch into a different function? I think `get_redshift_credentials` should return a dict that looks like
```
{
  "dbname": "...",
  "user": "...",
  "pass": "...",
  "port": "..."
}
```

regardless of if the auth method is credentials or IAM

Reply:
Done.